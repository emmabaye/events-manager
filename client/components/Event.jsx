import React, { Component } from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import DeleteModal from './DeleteModal.jsx'; //eslint-disable-line

/**
 * React  component for event card to be displayed
 * as "myevents" for authenticated regular user
 */
export default class Event extends Component {
  /**
   * React's method to render react component.
   * Renders card for event details
   *
   * @return {object}
   */
  render() {
    return (
      <div className="col-md-4 event" style={{ height: "300px" }}>
        <div className="card">
          <div className="card-block" style={{ height: "300px" }}>
            <h5 className="card-title"><b>{this.props.eventDetails.title}</b></h5>
            <p className="card-text h6">{`${this.props.eventDetails.description.substr(0, 50)}...`}</p>
            <p className="card-text h6"><b>Venue</b> {`${this.props.eventDetails.venue.substr(0, 25)}...`}</p>
            <p className="card-text h6">
              <b>Date</b>: {moment(this.props.eventDetails.startDate).format('DD MMMM YYYY')}
            </p>
            <p className="card-text h6"><b>Time</b>: {this.props.eventDetails.time}</p>
            <p className="text-center">
              <Link to={`/events/${this.props.eventDetails.id}`} className="btn btn-sm btn-primary">
                <i className="fa fa-info-circle fa-lg" /> View Event
              </Link>
              <Link to={`./modifyevent/${this.props.eventDetails.id}`} className="btn btn-sm btn-success">
                <i className="fa fa-edit fa-lg" /> Modify
              </Link>
              <Link to="#" className="btn btn-sm btn-danger" data-toggle="modal"
                data-target={`#${this.props.eventDetails.id}`}>
                <i className="fa fa-trash-o fa-lg" /> Delete
              </Link>
            </p>
          </div>
        </div>
        <DeleteModal item="event" itemId={this.props.eventDetails.id} />
      </div>
    );
  }
}
