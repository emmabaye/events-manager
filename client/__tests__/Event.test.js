import Event from '../components/Event.jsx';

describe('Event Component', () => {
  const props = {
    eventDetails: {
      id: '1',
      title: 'Seminar',
      description: {
        substr: () => {}
      },
      venue: {
        substr: () => {}
      },
      time: "5:00",
      date: '2020-01-01',
      image: "",
      Center: []
    }
  };

  describe('Event  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<Event {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
