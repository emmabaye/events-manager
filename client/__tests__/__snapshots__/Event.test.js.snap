// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component Event  should render self it should render for dumb component 1`] = `
<div
  className="col-md-4 event"
  style={
    Object {
      "height": "300px",
    }
  }
>
  <div
    className="card"
  >
    <div
      className="card-block"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <h5
        className="card-title"
      >
        <b>
          Seminar
        </b>
      </h5>
      <p
        className="card-text h6"
      >
        undefined...
      </p>
      <p
        className="card-text h6"
      >
        <b>
          Venue
        </b>
         
        undefined...
      </p>
      <p
        className="card-text h6"
      >
        <b>
          Date
        </b>
        : 
        16 July 2018
      </p>
      <p
        className="card-text h6"
      >
        <b>
          Time
        </b>
        : 
        5:00
      </p>
      <p
        className="text-center"
      >
        <Link
          className="btn btn-sm btn-primary"
          replace={false}
          to="/events/1"
        >
          <i
            className="fa fa-info-circle fa-lg"
          />
           View Event
        </Link>
        <Link
          className="btn btn-sm btn-success"
          replace={false}
          to="./modifyevent/1"
        >
          <i
            className="fa fa-edit fa-lg"
          />
           Modify
        </Link>
        <Link
          className="btn btn-sm btn-danger"
          data-target="#1"
          data-toggle="modal"
          replace={false}
          to="#"
        >
          <i
            className="fa fa-trash-o fa-lg"
          />
           Delete
        </Link>
      </p>
    </div>
  </div>
  <Connect(DeleteModal)
    item="event"
    itemId="1"
  />
</div>
`;
